rubicon_ml/client/mixin.py

```diff
@@ -410,15 +404,16 @@ class TagMixin:
     """Adds tag support to a client object."""
 
     def _get_taggable_identifiers(self):
-        dataframe_id = None
+        project_name, experiment_id = self._parent._get_identifiers()
+        entity_id = None
 
-        if isinstance(self, client.Dataframe):
-            project_name, experiment_id = self.parent._get_parent_identifiers()
-            dataframe_id = self.id
+        # experiments are not required to return an entity ID - they are the entity
+        if isinstance(self, client.Experiment):
+            experiment_id = self.id
```

Comment:
what is experiment_id if this line doesn't execute?

Reply:
its the second value returned from `self._parent._get_identifiers()` on line 407. in this case, `None` specifically

it needs this experiment case because `get_identifiers` on a project (a project is always the experiment's parent) will return `None` for `experiment_id` since a project of course doesn't have an experiment ID

but, this function still needs the experiment's ID, so it just gets it from `self.id`. any other class that we're tagging that requires an experiment ID will have it returned from `self._parent._get_identifiers()` before we get into the if-else as those will all have experiments for parents, not projects

Reply:
sounds good!